android/src/main/java/com/swmansion/reanimated/nodes/ParamNode.java

```diff
@@ -17,31 +19,40 @@ public ParamNode(int nodeID, ReadableMap config, NodesManager nodesManager) {
 
   @Override
   public void setValue(Object value) {
-    Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
-    String callID = mUpdateContext.callID;
-    mUpdateContext.callID = mPrevCallID;
-    ((ValueNode) node).setValue(value);
-    mUpdateContext.callID = callID;
+    try {
+      Node node = mNodesManager.findNodeById(mArgsStack.peek(), Node.class);
+      String callID = mUpdateContext.callID;
+      mUpdateContext.callID = mPrevCallID;
+      ((ValueNode) node).setValue(value);
+      mUpdateContext.callID = callID;
+    } catch (EmptyStackException e) {
+      throw new JSApplicationCausedNativeException(getClass().getSimpleName() + " is trying to set value with no context.", e);
+    }
   }
 
+  @Override
   public void beginContext(Integer ref, String prevCallID) {
     mPrevCallID = prevCallID;
     mArgsStack.push(ref);
   }
 
-
+  @Override
   public void endContext() {
     mArgsStack.pop();
```

Comment:
Is this vital?
If not `callback` will work in `proc`

Reply:
yes

Reply:
if you need this you can make a copy of the stack and pass it to the callback and the on resolving revert previous stack for a while?

Reply:
What if the `param` is a `proc`? wouldn't I need to do that for every param in the tree that is needed for the evaluation?

Reply:
Let's continue this [here](https://github.com/software-mansion/react-native-reanimated/pull/469#issuecomment-555013397)